### PR TITLE
Bug 1035144 - Modify signatures for AsyncShutdownTimeout to represent the shutdown conditions instead of the stack.

### PR DESCRIPTION
**Pull request by bsmedberg**
This is UNTESTED. It also doesn't have any unit tests in test_signature_utilities.py because I couldn't figure out how to set up the test data. I'd appreciate help finishing this up since I don't have a socorro dev environment handy.

### CODE REVIEW

File: socorro/processor/signature_utilities.py [hunk 1]
```diff
@@ -697,6 +698,40 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
...
         return result
 
+class SignatureShutdownTimeout(SignatureGenerationRule):
+    """Customize the signature for shutdown timeouts"""
+
+    def version(self):
+        return '1.0'
+
+    def _predicate(self, raw_crash, raw_dumps, processed_crash, proc_meta):
+        return 'AsyncShutdownTimeout' in raw_crash
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: Shouldn't that be `raw_crash.get('AsyncShutdownTimeout')` to also capture if it's there but empty?

**Comment by adngdb** [comment]
I have started reviewing this, and while doing so realized that we have been doing something stupid with another of our Rules for quite some time. I'll have a PR to your branch tomorrow.

**Comment by adngdb** [comment]
(Note: it has an impact on this branch as well, of course. )

**Comment by adngdb** [comment]
Alright, I have addressed the issues I found in https://github.com/bsmedberg/socorro/pull/2. @peterbe do you want to review this from here on?

**Comment by peterbe** [comment]
@adngdb Good to merge?

**Comment by adngdb** [comment]
Yes, let's land this. Thank you for your work on this @bsmedberg and @peterbe !

**Pull request merged**